src/Learning/KWData/KWClass.cpp

```diff
@@ -2660,9 +2659,19 @@ void KWClass::InitializeRandomRuleParameters(KWDerivationRule* rule, const ALStr
 	}
 }
 
-boolean KWClass::CheckClassComposition(KWAttribute* parentAttribute, NumericKeyDictionary* nkdComponentClasses) const
+boolean KWClass::CheckNativeComposition(boolean bCheckKeys, boolean bVerboseCheckKeys) const
+{
+	NumericKeyDictionary nkdComponentClasses;
```

Comment:
Why do we need to pass (pointer to) a `NumericKeyDictionary` object, rather than create it directly in `InternalCheckNativeComposition`, where it is updated _and_ used / checked?

Reply:
Commentaire ajouté (bien qu'un peu redondant dans ce cas mono-instruction avec les commentaires de la méthode appelée)

Reply:
Ok.